core/jvm/src/main/scala/scalapb/zio_grpc/ZCall.scala

```diff
@@ -0,0 +1,44 @@
+package scalapb.zio_grpc
+
+import io.grpc.Status
+import scalapb.zio_grpc.ZCall.ReadyPromise
+import zio.{Promise, Ref, Semaphore, UIO, ZIO}
+
+object ZCall {
+  type ReadyPromise = Ref[Option[Promise[Nothing, Unit]]]
+}
+
+trait ZCall[-R, A] extends Any {
```

Comment:
Since there's a possibility that this change would cause issues to some users or result in throughput degradation, I want to see if we can make this change such that there's one `ZCall` implementation that supports backpressue and another `ZCall` that behaves like the previous one, where all the construction and wiring is done here? No need to make it configurable for the user, but I want to be able to swap it out during development as things stabilize.